tests/page-objects/default/tasks/tasks.wdio.page.js

```diff
@@ -9,20 +9,47 @@ const FORM_TITLE_SELECTOR = `${TASK_FORM_SELECTOR} h3#form-title`;
 const NO_SELECTED_TASK_SELECTOR = '.empty-selection';
 
 const getTaskById = (emissionId) => $(`${TASK_LIST_SELECTOR} li[data-record-id="${emissionId}"`);
-const getTasks = () => $$(`${TASK_LIST_SELECTOR} li.content-row`);
+const getTasks = async () => {
+  await browser.waitUntil(async () => {
+    const tasksList = await $(TASK_LIST_SELECTOR);
+    return await tasksList.isDisplayed();
+  }, {
+    timeout: 10000,
+    timeoutMsg: 'Expected tasks list to be displayed'
+  });
+
+  await browser.waitUntil(async () => {
+    const tasks = await $$(`${TASK_LIST_SELECTOR} li.content-row`);
+    if (!tasks.length) {
+      return false;
+    }
+    for (const task of tasks) {
+      if (!(await task.isDisplayed())) {
+        return false;
+      }
+    }
+    return true;
+  }, {
+    timeout: 10000,
+    interval: 500,
+    timeoutMsg: 'Expected task elements to be present and displayed'
+  });
```

Comment:
Can this be pulled out into its own function "waitForTaskLoading" ?

Reply:
I’ve already simplified it significantly based on your earlier suggestion to remove the redundant isDisplayed() checks on individual tasks. Additionally, I combined the two waitUntil blocks into one. Let me know what you think.